packages/commons/src/interceptors/consistencyHandling.ts

```diff
@@ -0,0 +1,39 @@
+import { AxiosResponseHeaders, AxiosInstance } from "axios";
+import ApiClientBase from "../core/ApiClientBase.js";
+
+export function configureConsistencyHandlingInterceptor(
+  axios: AxiosInstance,
+): void {
+  let lastEventId: string | undefined = undefined;
+
+  axios.interceptors.request.use((config) => {
+    const isMutatingRequest = ["post", "put", "delete", "patch"].includes(
+      config.method?.toLowerCase() ?? "",
+    );
+    if (lastEventId !== undefined && !isMutatingRequest) {
+      config.headers["if-event-reached"] = lastEventId;
+    }
+    return config;
+  });
+
+  axios.interceptors.response.use((response) => {
+    const isMutatingRequest = ["post", "put", "delete", "patch"].includes(
+      response.config.method?.toLowerCase() ?? "",
+    );
```

Comment:
Duplicate code (greetings Uncle Bob)

Reply:
Oh. I was sure I had extracted this to a function … 🙈